src/server/images/controllers.js

```diff
@@ -0,0 +1,61 @@
+import {ENTITY, STATE, KeyUtils} from '../key-utils';
+import {S3Connector} from '../aws-s3';
+import {APIError} from '../ErrorHandler';
+import ImageManager from './models';
+
+export default {
+  get(req, res, cb) {
+    const {itemid, imageid} = req.query;
+    if (itemid) {
+      // get all images of item
+      const keys = [];
+      const promises = [];
+      const checkState = [STATE.ALIVE, STATE.EXPIRED];
+
+      for (const state of checkState) {
+        promises.push(new Promise((resolve, reject) => {
+          const prefix = KeyUtils.getPrefix(ENTITY.IMAGE, state, itemid);
+          ImageManager.fetchPrefix(prefix, (err, data) => {
+            if (err) {
+              reject(err);
+              return;
+            }
+            data.map((key) => {
+              keys.push(key);
+            });
+            resolve();
```

Comment:
We get origin image's keys in data arr using this fetchPrefix function.
We have to copy element's of data array to keys array?
I think we can just 'resolve(data)' to send keys array to .then(keys) block.